packages/aws-cdk-lib/aws-ecs/lib/cluster.ts

```diff
@@ -263,6 +290,37 @@ export class Cluster extends Resource implements ICluster {
     Aspects.of(this).add(new MaybeCreateCapacityProviderAssociations(this, id));
   }
 
+  private updateKeyPolicyForEphemeralStorageConfiguration(key: IKey, clusterName?: string) {
+    const clusterConditions: PolicyStatementProps['conditions'] = {
+      StringEquals: {
+        'kms:EncryptionContext:aws:ecs:clusterAccount': [Aws.ACCOUNT_ID],
+      },
+    };
+    if (clusterName) {
+      clusterConditions.StringEquals['kms:EncryptionContext:aws:ecs:clusterName'] = [clusterName];
+    }
```

Comment:
```suggestion
    const clusterConditions = {
      StringEquals: {
        'kms:EncryptionContext:aws:ecs:clusterAccount': [Aws.ACCOUNT_ID],
        ...(clusterName && { 'kms:EncryptionContext:aws:ecs:clusterName': [clusterName] }),
      },
    };
```
`PolicyStatementProps['conditions']` is a [generic object](https://github.com/aws/aws-cdk/blob/da2ec75f6bc8f2e28b07cdf1307b22c83bb652d5/packages/aws-cdk-lib/aws-iam/lib/policy-statement.ts#L787) so we don't gain much information by specifying it.
Also, we could keep it a bit cleaner.